File: app/src/main/java/br/com/nicetuber/feature/channel_details/IChannelDetails.java
package br.com.nicetuber.feature.channel_details;

import br.com.nicetuber.base.IView;
import br.com.nicetuber.model.ChannelStatistics;

/**
 * Created by thiagozg on 15/10/2017.
 */

public interface IChannelDetails {

    interface View extends IView {
        void showChannelStatistics(ChannelStatistics channelStatistics);
        void showErrorMessage(String message);
    }

    interface Callback {
        void getChannelStatisticsSuccess(ChannelStatistics channelStatistics);
        void getChannelStatisticsError(String message);
    }

}

File: app/src/main/java/br/com/nicetuber/feature/channel_details/IChannelDetails.kt
package br.com.nicetuber.feature.channel_details

import br.com.nicetuber.base.IView
import br.com.nicetuber.model.ChannelStatistics

/**
 * Created by thiagozg on 15/10/2017.
 */

interface IChannelDetails {

    interface View : IView {
        fun showChannelStatistics(channelStatistics: ChannelStatistics)
        fun showErrorMessage(message: String)
    }

    interface Callback {
        fun getChannelStatisticsSuccess(channelStatistics: ChannelStatistics)
        fun getChannelStatisticsError(message: String)
    }

}
